flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -348,6 +348,18 @@
 				+ TaskManagerOptions.MEMORY_OFF_HEAP.key() + "', to use either all on-heap memory or all off-heap memory"
 				+ " for Managed Memory.");
 
+	/**
+	 * Off-Heap Managed Memory size.
+	 * This config option should only be used for passing derived off-heap managed memory size into TaskExecutors. It
+	 * should not be exposed to users.
+	 */
+	@Documentation.ExcludeFromDocumentation
+	public static final ConfigOption<String> MANAGED_MEMORY_OFFHEAP_SIZE =
+		key("taskmanager.memory.managed.off-heap.size")
+			.noDefaultValue()
+			.withDescription("DO NOT USE THIS CONFIG OPTION! This is the config option for Off-Heap Managed Memory size."
```

Comment:
Why shouldn't it be used? Can't we do a sanity check and fail if things don't add up?